internal/core/data/errors/types.go

```diff
@@ -150,3 +150,15 @@ func (e ErrInvalidId) Error() string {
 func NewErrInvalidId(id string) error {
 	return &ErrInvalidId{id: id}
 }
+
+type ErrInvalidName struct {
```

Comment:
Since this new error type is used within core-command, it should be declared in the internal/core/command package. It's not a good practice to have imports straddle from one service to the other as found in restDevice.go
`import "github.com/edgexfoundry/edgex-go/internal/core/data/errors"`